File: clash/modify-script/modify-squirrelvpn.go
package main

import (
  "io/ioutil"
  "log"

  "gopkg.in/yaml.v3"
)

func main() {

  // Load original YAML
  dataBytes, err := ioutil.ReadFile("../squirrelvpn.yaml")
  if err != nil {
    log.Fatal(err)
  }

  var data map[string]interface{}
  err = yaml.Unmarshal(dataBytes, &data)
  if err != nil {
    log.Fatal(err)
  }

  // Load proxy YAML
  proxyBytes, err := ioutil.ReadFile("proxy.yaml")
  if err != nil {
    log.Fatal(err)
  }

  var proxyConfigs []interface{}
  err = yaml.Unmarshal(proxyBytes, &proxyConfigs)
  if err != nil {
    log.Fatal(err)
  }

  // Append proxies
  proxies, ok := data["proxies"].([]interface{})
  if !ok {
    log.Fatal("proxies not found")
  }

  data["proxies"] = append(proxies, proxyConfigs...)

  // Load rules YAML
  rulesBytes, err := ioutil.ReadFile("rules.yaml")
  if err != nil {
    log.Fatal(err)
  }

  var rulesConfigs []interface{}
  err = yaml.Unmarshal(rulesBytes, &rulesConfigs)
  if err != nil {
    log.Fatal(err)
  }

  // Prepend rules
  rules, ok := data["rules"].([]interface{})
  if !ok {
    log.Fatal("rules not found")
  }

  data["rules"] = append(rulesConfigs, rules...)

  // Add proxy to groups
  proxyGroups, ok := data["proxy-groups"].([]interface{})
  if !ok {
    log.Fatal("proxy-groups not found")
  }

  proxyGroup := proxyGroups[0].(map[string]interface{})
  groupProxies := proxyGroup["proxies"].([]interface{})
  proxyGroup["proxies"] = append(groupProxies, "shadowsocks")

  // Marshal and write output
  output, err := yaml.Marshal(data)
  if err != nil {
    log.Fatal(err)
  }

  err = ioutil.WriteFile("../squirrelvpn-modified.yaml", output, 0644)
  if err != nil {
    log.Fatal(err)
  }
}

File: clash/modify-script/python/modify-squirrelvpn.py
# Modify the config file of SquirrelVPN to add shadow-socks proxy and bypass some site like openai.com and claude.ai\
# How to use:
# pip install pyyaml
# python modify-squirrelvpn.py
import yaml


class MyDumper(yaml.Dumper):

    def increase_indent(self, flow=False, indentless=False):
        return super(MyDumper, self).increase_indent(flow, False)


# Load original squirrelvpn.yaml
with open('../../squirrelvpn.yaml', encoding='utf8') as f:
    data = yaml.safe_load(f)

# Load proxy.yaml containing shadowsocks proxy config    
with open('../config/proxies.yaml', encoding='utf8') as f:
    proxies = yaml.safe_load(f)

# Append imported proxy to proxies list
# append all proxies, not the first one
data['proxies'].append(*proxies['proxies'])

# Load rules.yaml containing routing rules
with open('../config/rules.yaml', encoding='utf8') as f:
    rules = yaml.safe_load(f)

# Insert rules to beginning of rules list
data['rules'][0:0] = rules['rules']

data["proxy-groups"][0]["proxies"].append("shadowsocks")

# Write out modified data to new yaml file 
with open('../../squirrelvpn-modified.yaml', 'w', encoding='utf8') as f:
    yaml.dump(data, f, Dumper=MyDumper, default_flow_style=False, allow_unicode=True)
